pumpio/pumpio.php

```diff
@@ -1042,21 +1042,15 @@ function pumpio_get_contact($uid, $contact, $no_insert = false) {
 function pumpio_dodelete(&$a, $uid, $self, $post, $own_id) {
 
 	// Two queries for speed issues
-	$r = q("SELECT * FROM `item` WHERE `uri` = '%s' AND `uid` = %d LIMIT 1",
-				dbesc($post->object->id),
-				intval($uid)
-		);
-
-	if (count($r))
-		return Item::deleteById($r[0]["id"]);
-
-	$r = q("SELECT * FROM `item` WHERE `extid` = '%s' AND `uid` = %d LIMIT 1",
-				dbesc($post->object->id),
-				intval($uid)
-		);
+	$condition = ['uri' => $post->object->id, 'uid' => $uid];
+	if (dba::exists('item', $condition)) {
+		return Item::delete($condition);
+	}
 
-	if (count($r))
-		return Item::deleteById($r[0]["id"]);
+	$condition = ['extid' => $post->object->id, 'uid' => $uid];
+	if (dba::exists('item', $condition)) {
```

Comment:
Same comment here, is there a viable case where we need to delete items with `uri = $post->object->id` but not `extid = $post->object->id`?

Reply:
It is either the one **or** the other.

Reply:
I got that, is there a specific reason to do this way?

Reply:
The ```uri``` will contain the pump.io specific value when it originally came from pump.io. When we created the post on Friendica then the ```uri``` will contain an ```uri``` in our format. But the ```extid``` will then contain the ```uri``` like it is on pump.io.

Reply:
So it’s fine to do both anyway since I can’t imagine a scenario where we would need to keep an item with a Pump.io `extid` but not the item with a Pump.io `uri`.

Reply:
This was one of these "never change a running system" things. I want to rework that whole addon - but not now.